### PR TITLE
configurator + config for rpc and keys dir

### PR DESCRIPTION
**Pull request by bobbyrauchenberg**

### CODE REVIEW

File: Config.hs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+nanocoin {
+  rpcPort = 3000
+  #keysPath = "/path/to/keys" (optional)
```
Comment (tdietert) [inline Config.hs]: What is this '#' character, and parentheses "optional"?

File: exe/Main.hs [hunk 2]
```diff
@@ -13,12 +17,17 @@ data Config = Config
...
+readConfig cfgFile = do
+  cfg <- C.load [C.Required cfgFile]
+  rp  <- C.require cfg "nanocoin.rpcPort"
+  kp  <- C.lookup  cfg "nanocoin.keysPath"
+  return $ Config rp kp
 
 main :: IO ()
 main = do
-    Config rpc mKeys <- execParser (info parser mempty)
+    fromConfig <- readConfig "Config.hs"
```
Comment (tdietert) [inline exe/Main.hs]: Can you add a comment "-- Read config from config file", above this line?

File: exe/Main.hs [hunk 3]
```diff
@@ -13,12 +17,17 @@ data Config = Config
...
+    Config rpc mKeys <- execParser (info (parser fromConfig) mempty)
```
Comment (tdietert) [inline exe/Main.hs]: Can you add a comment "-- Override config with cmd line arguments" above this line?

**Comment by bobbyrauchenberg** [comment]
I've made the changes requested in https://github.com/tdietert/nanocoin/pull/27. There were a few messy commits and wasn't sure what the deal was about squashing, so i've just pushed this on another branch with the changes requested in the PR https://github.com/tdietert/nanocoin/pull/27

**Pull request closed without merging**